logger.go

```diff
@@ -51,10 +57,22 @@ func (l *Logger) SetWriter(w io.Writer) {
 	l.mu.Unlock()
 }
 
+func (l *Logger) SetFlags(flags int32) {
+	l.mu.Lock()
+	l.flags = l.flags | flags
+	l.mu.Unlock()
+}
+
 func (l *Logger) format(buffer *bytes.Buffer, lline logLine) {
 	var dynamicContext C
 	now := time.Now()
-	fmt.Fprintf(buffer, prefixFormat, now.Format(timeFormat), levelNames[lline.level])
+
+	var flagsFields string
+	if l.flags&(Llongfile|Lshortfile|Lmethod) != 0 {
```

Comment:
I think l.flags shouldn't be read without acquiring the mutex first  ... An even better alternative could be use an atomic int32 as we do with `level`